ui/src/app/base/components/AuthenticationFields/AuthenticationFields.tsx

```diff
@@ -0,0 +1,86 @@
+import { Button, Icon, Input, Textarea } from "@canonical/react-components";
+
+import FormikField from "app/base/components/FormikField";
+
+type Props = {
+  certificateValueName?: string;
+  onRadioChange: () => void;
```

Comment:
Do you think this would be better as a function `onShouldGenerateCert: (shouldGenerateCert: boolean) => void;` and then in PowerTypeFields.tsx you can respond with the boolean rather than toggling the state? It just might be a bit more reliable than just changing the value when the radio changes.

Reply:
Yeah I think you're right, `onRadioChange` is also a bit non-descriptive for a prop name